internal/usecase/repository/hostconf_jwk_repository.go

```diff
@@ -105,12 +105,14 @@ func (r *hostconfJwkRepository) PurgeExpiredJWKs(db *gorm.DB) (hcjwks []model.Ho
 		slog.Error(err.Error())
 		return nil, err
 	}
-	if err = db.
-		Unscoped(). // do not use GORM's soft delete for purging
-		Delete(&hcjwks).
-		Error; err != nil {
-		slog.Error(err.Error())
-		return nil, err
+	if len(hcjwks) > 0 {
+		if err = db.
+			Unscoped(). // do not use GORM's soft delete for purging
+			Delete(&hcjwks).
+			Error; err != nil {
+			slog.Error(err.Error())
+			return nil, err
+		}
```

Comment:
Meanwhile debugging I got some situations when no keys exists and executing the `Delete` operation with an empty slice, so only execute `Delete` when there is something to delete.